contracts/smart-wallet/src/webauthn.rs

```diff
@@ -2,16 +2,45 @@ use soroban_sdk::{contracttype, crypto::Hash, panic_with_error, Bytes, BytesN, E
 
 use crate::{base64_url, AccountContractError};
 
+/// The WebAuthn type for the get operation.
+pub(crate) const WEBAUTHN_TYPE_GET: &str = "webauthn.get";
+
 /// Authenticator data flag offset. It appears after the RP ID hash in the authenticator data.
 pub(crate) const AUTH_DATA_FLAG_OFFSET: u32 = 32;
 /// Authenticator data flags for user presence
 pub(crate) const AUTH_DATA_FLAG_UP: u8 = 0x01;
 /// Authenticator data flags for user verification
 pub(crate) const AUTH_DATA_FLAG_UV: u8 = 0x04;
+
 /// Length of the encoded challenge in the client data JSON.
 pub(crate) const ENCODED_CHALLENGE_LEN: u32 = 43;
-/// The WebAuthn type for the get operation.
-pub(crate) const WEBAUTHN_TYPE_GET: &str = "webauthn.get";
+
+/// Max length of the client data JSON in bytes.
+///
+/// #### Explanation of the length:
+///
+/// - `type`: ~20 bytes (`"type":"webauthn.get"`).
+///
+/// - `challenge`: ~43 bytes (`"challenge":"<base64url_32_byte_challenge>",`).
+///
+/// - `origin`: ~100-200 bytes (`"origin":"https://example.com",`)
+///
+/// - `crossOrigin`: ~20 bytes (`"crossOrigin":false,`).
+///
+/// Total length: ~283 bytes.
+///
+/// This is a conservative estimate, as the actual length may vary based on the specific values used.
+/// The maximum length is set to 1024 bytes to accommodate any additional fields or whitespace.
+pub(crate) const MAX_CLIENT_DATA_JSON_LEN: usize = 1024;
```

Comment:
Seems reasonable to me. 

I'm wondering what we could do to take away the need for this. I think the explicit size assumption could be removed if the to_buffer function was accompanied by a to_reader function that provided access to reading the string in chunks, and if serde_json_core supported accepting a read-like interface. Unfortunately serde_json_core doesn't support anything other than a full slice or string.